### DISCUSSION TITLE
Bit 3 low address byte

### DISCUSSION THREAD

**Discussion by mwr07071** [thehighnibble/thehighnibble.github.io#150] [IMSAI8080esp replica]
After a few weeks of trouble-free operation, bit 3 of the low order address byte isn't responding properly to examine.
The led seems fine, lights up with a reset.  And the switch seems to work, but in reverse:
1.  Flip switch 3 up on low address byte, then examine.  The led doesn't light.
2. Flip switch 3 down (don't raise examine) and the led lights.

When I deposit an 8 into address 8, then single step this code:
.org 	128
	lxi		h,8
	mov		a,m
	cma
	out 	0ffh
	.end
Programmed output shows bit 3 lit.
When I deposit a zero into address 8 and single step the code, programmed output shows bit 3 off as expected.

So even though the led and switch are working, and the switch still works correctly for accessing and setting memory with a deposit, the state of the led in direct response to the switch/examine combination for this one address bit is visually wrong.

Suggestions would be appreciated.

**Comment by nbreeden2** [comment]
I think you've already provided this info but it's just not jumping out at me.
When you single step through code does LED-3 act correctly? Is it showing the correct address?
When you step through addresses with "examine next" does LED-3 act correctly? Is it showing the correct address?
How about with "deposit next"?

One thought would be to swap the 595 IC that drives bit 3 with another that works and see if the issue follows it. The traces (LED to chip) on the PCB will lead you to the correct IC. This may be tricky as the 595 may be under the switch frames.

**Reply by mwr07071** [reply]
I filled address 0 thru F with zeros.  As I used deposit next to store each byte, bit 3 never lit.
When single-stepping thru those nops, bit 3 never lit.
Same thing with using examine next.

**Reply by nbreeden2** [reply]
Thanks for the info.

One last test. When you hold up RESET am I correct the LED doesn't light?

This issue is pretty odd. As you've indicated the LED does light (on for 0, off for 1) so the only thing I can think of are the 74xx595s that drive the LEDs. You can swap two of then to see if the issue follows the IC. If you decide to try to swap two of them then I believe the 595 for A7..A0 is under the metal switch tabs and very difficult to access. 

Unless you have good experience desoldering components like switches and the proper tools the odds of damaging the PCB are very high. I can't advise if you should try this or not.

I'm told you can also bend up the tabs on the switches to provide clearance, I haven't tried this.

If you're in the US and want someone to take a look / do any required repair I may be able to do this for you. I have the tools and expertise. I know Dave (the High Nibble) and he can vouch for me.

**Comment by mwr07071** [comment]
When I hold up reset, bit 3 does not light.  I'm in the Denver area.

**Comment by nbreeden2** [comment]
Just to confirm: You're running the IMSAI firmware, not the Cromemco firmware?

I'm about 3 hours north of Denver in Torrington WY.

**Comment by mwr07071** [comment]
Hi Neil,

Yes - I'm running IMSAI firmware.
I'd be glad to pay for FedEx shipping.

Regards,
Mark Rosenthal

**Comment by nbreeden2** [comment]
email me at  nbreeden@me.com  and we can make arrangements.
